src/hooks/useAspectCreatedOnCalloutSubscription.ts

```diff
@@ -0,0 +1,101 @@
+import { useApolloErrorHandler } from './index';
+import {
+  usePrivilegesOnHubCollaborationQuery,
+  useHubAspectsQuery,
+  usePrivilegesOnChallengeCollaborationQuery,
+  useChallengeAspectsQuery,
+  usePrivilegesOnOpportunityCollaborationQuery,
+  useOpportunityAspectsQuery,
+} from './generated/graphql';
+import { AspectCardFragment, AuthorizationPrivilege, Scalars } from '../models/graphql-schema';
+import useCalloutAspectCreatedSubscription from '../domain/collaboration/useCalloutAspectCreatedSubscription';
+
+export type AspectWithPermissions = AspectCardFragment & { canDelete: boolean | undefined };
+export interface AspectsData {
+  subscriptionEnabled: boolean;
+}
+
+interface UseAspectDataHookProps {
+  hubNameId: Scalars['UUID_NAMEID'];
+  calloutId: Scalars['UUID_NAMEID'];
+  challengeNameId?: Scalars['UUID_NAMEID'];
+  opportunityNameId?: Scalars['UUID_NAMEID'];
+}
+
+export const useAspectCreatedOnCalloutSubscription = ({
+  hubNameId,
+  calloutId,
+  challengeNameId = undefined,
+  opportunityNameId = undefined,
+}: UseAspectDataHookProps): AspectsData => {
+  const handleError = useApolloErrorHandler();
+
+  const { data: hubCollaborationData } = usePrivilegesOnHubCollaborationQuery({
+    variables: { hubNameId },
+    skip: !!(challengeNameId || opportunityNameId),
+    onError: handleError,
+  });
+  const hubCollaboration = hubCollaborationData?.hub?.collaboration;
+  const hubCollaborationPrivileges = hubCollaboration?.authorization?.myPrivileges;
+  const canReadHubCollaboration = hubCollaborationPrivileges?.includes(AuthorizationPrivilege.Read);
+  const { data: hubAspectData, subscribeToMore: subscribeToHub } = useHubAspectsQuery({
```

Comment:
You don't need all callouts with all aspects fetched.

Reply:
Nice catch, fixing. I actually don't need any aspects.